### PR TITLE
lighttpd: update to lighttpd 1.4.69 release hash

### PR DESCRIPTION
**Pull request by gstrauss**
Maintainer: @flyn-org
Compile tested: arm_cortex-a9 OpenWrt master

Description:

    update to lighttpd 1.4.69 release hash
    add lighttpd-mod-webdav_min package
        lighttpd-mod-webdav_min is more minimal than full lighttpd-mod-webdav.
        lighttpd-mod-webdav_min does not support PROPPATCH, LOCK, UNLOCK, and
        by not supporting those methods, removes dependencies on libxml2,
        libsqlite3, and libuuid.

### CODE REVIEW

**Comment by gstrauss** [comment]
aarch64-cortex-a53 and x86_64 builds fail with error on dependency, not due to this PR:
```
Collected errors:
 * pkg_hash_check_unresolved: cannot find dependency kernel (= 5.15.93-1-b0eafca757efb883a35dbbcaf514bbc8) for kmod-fs-autofs4
 * satisfy_dependencies_for: Cannot satisfy the following dependencies for pgsql-server:
 * 	kernel (= 5.15.93-1-b0eafca757efb883a35dbbcaf514bbc8)
 * opkg_install_cmd: Cannot install package pgsql-server.
```

**Comment by gstrauss** [comment]
@flyn-org @BKPepe please review and approve.  Thank you.

**Comment by gstrauss** [comment]
rebased to tip of master in order to force CI rebuild.

File: net/lighttpd/patches/020-meson-mod_webdav_min.patch [hunk 1]
```diff
@@ -0,0 +1,28 @@
+From fe6895e5eb2c17ddad6e95faaac7d56d4dd73256 Mon Sep 17 00:00:00 2001
+From: Glenn Strauss <gstrauss@gluelogic.com>
+Date: Fri, 20 Jan 2023 18:48:14 -0500
+Subject: [PATCH] [meson] mod_webdav_min w/o deps: xml2 sqlite3 uuid
```
[the remark this answers is not in the document]
Comment (gstrauss) [reply net/lighttpd/patches/020-meson-mod_webdav_min.patch:4]: As suggested, I updated and expanded the description.

**Comment by gstrauss** [comment]
<sigh> It appears that the build of openssl is failing on multiple platforms, preventing the PR from building.

**Comment by Ansuel** [comment]
@cotequeiroz ?

**Comment by cotequeiroz** [comment]
The engines-1.1 path tells me that `include/openssl_engine.mk` is not up-to-date:
```
install -m0755  /home/build/openwrt/build_dir/target-mips_24kc_musl/openssl-3.0.8/ipkg-install/usr/lib/engines-1.1/afalg.so /home/build/openwrt/build_dir/target-mips_24kc_musl/openssl-3.0.8/ipkg-mips_24kc/libopenssl-afalg/usr/lib/engines-1.1
```
Is it because of an old SDK?

**Comment by cotequeiroz** [comment]
All of the SKD have been updated with the new version of `include/openssl_engine.mk`.  I've just triggered new checks here to see if it indeed work as expected.  The binary packages will need some more time before they are all built with the new SDK.  I checked some of the latest builds, and they appear to be fine--the engine packages were the ones failing; you may see the main libopessl3 package in failed attempts, but if the engines are there, it went through.

**Comment by cotequeiroz** [comment]
It is running now, and most of the targets are building OK now.  It was important to me to detect any problems, and I do have work to do with arc and ppc.  Since it seems it will not take too long, I'll wait until they're all done and merge this.

Sorry about the noise here.

**Comment by gstrauss** [comment]
Tests completed now, but 4 platforms in the CI tests still fail on openssl.

2 missing libatomic.so.1.
```
Package libopenssl is missing dependencies for the following libraries:
libatomic.so.1
```

1 missing symbols from libataomic:

```
/home/build/openwrt/staging_dir/toolchain-arc_archs_gcc-12.2.0_glibc/bin/../lib/gcc/arc-openwrt-linux-gnu/12.2.0/../../../../arc-openwrt-linux-gnu/bin/ld: ./libcrypto.so: undefined reference to `__atomic_is_lock_free'
/home/build/openwrt/staging_dir/toolchain-arc_archs_gcc-12.2.0_glibc/bin/../lib/gcc/arc-openwrt-linux-gnu/12.2.0/../../../../arc-openwrt-linux-gnu/bin/ld: ./libcrypto.so: undefined reference to `__atomic_fetch_or_8'
/home/build/openwrt/staging_dir/toolchain-arc_archs_gcc-12.2.0_glibc/bin/../lib/gcc/arc-openwrt-linux-gnu/12.2.0/../../../../arc-openwrt-linux-gnu/bin/ld: ./libcrypto.so: undefined reference to `__atomic_load_8'
```

1 failed lighttpd CI test via docker container:

```
Collected errors:
 * pkg_hash_check_unresolved: cannot find dependency libopenssl3 for lighttpd-mod-openssl
 * pkg_hash_fetch_best_installation_candidate: Packages for lighttpd-mod-openssl found, but incompatible with the architectures configured
 * opkg_install_cmd: Cannot install package lighttpd-mod-openssl.```

**Comment by cotequeiroz** [comment]
The PPC and ARC failures are my work-to-do.  aarch64-cortexa53 is failing during run-time because phase2 bots have not built the new openssl 3.0 packages yet.  None of this is related to this PR.  I'll merge this now.

**Comment by flyn-org** [comment]
Thank you, @gstrauss  and @cotequeiroz!

**Pull request merged**